contracts/contracts/poolBooster/PoolBoosterMetropolis.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { IPoolBooster } from "../interfaces/poolBooster/IPoolBooster.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+ * @title Pool booster for Metropolis pools
+ * @author Origin Protocol Inc
+ */
+contract PoolBoosterMetropolis is IPoolBooster {
+    // @notice address of the OS token
+    IERC20 public immutable osToken;
+    // @notice address of the pool
+    address public immutable pool;
+    // @notice if balance under this amount the bribe action is skipped
+    uint256 public constant MIN_BRIBE_AMOUNT = 1e10;
+
+    IRewarderFactory public immutable rewardFactory;
+
+    IVoter public immutable voter;
+
+    constructor(
+        address _osToken,
+        address _rewardFactory,
+        address _pool,
+        address _voter
+    ) {
+        require(_pool != address(0), "Invalid pool address");
+        pool = _pool;
+        // Abstract factory already validates this is not a zero address
+        osToken = IERC20(_osToken);
+
+        rewardFactory = IRewarderFactory(_rewardFactory);
+
+        voter = IVoter(_voter);
+    }
+
+    function bribe() external override {
+        uint256 balance = osToken.balanceOf(address(this));
+        // balance too small, do no bribes
+        (, uint256 minBribeAmount) = rewardFactory.getWhitelistedTokenInfo(
+            address(osToken)
+        );
+        if (balance < MIN_BRIBE_AMOUNT || balance < minBribeAmount) {
+            return;
```

Comment:
nit: you could throw an exception here instead of silently succeeding just so when we do it manually it is more obvious

Reply:
I've done it in this way to avoid having `bribeAll` failing to often for small amount. And having it silently succeeding is more aligned with the other PoolBooster implementation. 
Is it ok for you if I keep it like this? If no I would change it ofc.

Reply:
yes that is a great point. Lets leave it like that